mono/metadata/assembly.c

```diff
@@ -1829,6 +1830,26 @@ mono_assembly_load_friends (MonoAssembly* ass)
 }
 
 
+struct HasReferenceAssemblyAttributeIterData {
+	gboolean has_attr;
+};
+
+static gboolean
+has_reference_assembly_attribute_iterator (MonoImage *image, const MonoAssemblyName *aname, const char *nspace, const char *name, gpointer user_data)
+{
+	gboolean continue_scanning = TRUE;
+	struct HasReferenceAssemblyAttributeIterData *iter_data = (struct HasReferenceAssemblyAttributeIterData*)user_data;
+
+	if (!strcmp (name, "ReferenceAssemblyAttribute") && !strcmp (nspace, "System.Runtime.CompilerServices")) {
+		if (!strcmp (aname->name, "System.Runtime") || !strcmp (aname->name, "mscorlib")) {
```

Comment:
This is not good. With type forwarders the type can be declared anywhere (e.g. netstandard will add another place)